x-pack/plugins/reporting/public/lib/reporting_client.ts

```diff
@@ -0,0 +1,38 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+import { kfetch } from 'ui/kfetch';
+
+// @ts-ignore
+import rison from 'rison-node';
+import chrome from 'ui/chrome';
+import { QueryString } from 'ui/utils/query_string';
+import { jobCompletionNotifications } from '../services/job_completion_notifications';
+
+const API_BASE_URL = '/api/reporting/generate';
+
+class ReportingClient {
+  public getReportingJobPath = (exportType: string, jobParams: any) => {
+    return `${chrome.addBasePath(API_BASE_URL)}/${exportType}?${QueryString.param(
+      'jobParams',
+      rison.encode(jobParams)
+    )}`;
+  };
+
+  public createReportingJob = (exportType: string, jobParams: any) => {
+    const query = {
+      jobParams: rison.encode(jobParams),
+    };
+    return kfetch({ method: 'POST', pathname: `${API_BASE_URL}/${exportType}`, query }).then(
```

Comment:
need to return here? or can we do `await kfetch`?  I don't see the return val being used anywhere.

Reply:
Yes, we need to return the promise returned from calling kfetch. That promise is used in reporting_panel_context to handle fetch exceptions. There is a `catch` block.

Reply:
Created https://github.com/elastic/kibana/issues/22790

Reply:
I think if you make this an async function and use await the returned promise is implicit.  Should still work with the catch.  E.g.

```
 const response = await kfetch({ method: 'POST', pathname: `${API_BASE_URL}/${exportType}`, query });
 jobCompletionNotifications.add(response.job.id);
```

I don't think you need to return the response, just the promise.